### PR TITLE
Fix dropdown textfield combobox spin button wcag issues

### PR DESCRIPTION
**Pull request by betrue-final-final**
These fixes addresses WCAG 2.1 non-text contrast guidelines https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast

I've updated the focus and pressed states of these components:

Textfield
Combobox
Dropdown
Spinbutton
Before and after
image
https://user-images.githubusercontent.com/20363037/64815800-42256d00-d55b-11e9-810e-57d0d9648a51.png

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10885)

### CODE REVIEW

**Comment by ecraig12345** [comment]
@betrue-final-final Just to be clear, which scenarios should we go through when testing this? I'm guessing 1) focus using mouse, 2) focus using keyboard, 3) repeat in high contrast.

**Comment by ecraig12345** [comment]
Also the build is failing now because of a missing change file for the date-time package :(

**Comment by betrue-final-final** [comment]
@ecraig12345 , I've been working with Jon Schechtman on this, and I've run yarn update-snapshots, and also ran yarn update-snapshots locally for date picker, so I'm not sure what's going on.

**Comment by betrue-final-final** [comment]
As for testing, it really is a visual change only to meet contrast requirements for focus and selection. I would just test that in HC and non-HC to make sure I didn't break HC.

**Comment by betrue-final-final** [comment]
So should I run yarn change locally for date time?

**Comment by betrue-final-final** [comment]
I just ran yarn change again, and this is what I get
=======================================
Your local repository already has change files for these packages: @uifabric/date-time, office-ui-fabric-react
No change files are needed
✨  Done in 1.35s.
betrueMacs-MBP:office-ui-fabric-react betruemac$

**Comment by joschect** [comment]
Closing and reopening to trigger rerun

**Comment by joschect** [comment]
Reopening to rebuild.

**Comment by ecraig12345** [comment]
This time it built, but some of the things flagged by screener look like actual regressions.

**Comment by ecraig12345** [comment]
Now it needs more snapshot updates, in the main package and possibly also date-time and experiments. :(

**Comment by ecraig12345** [comment]
Just pushed an update with additional snapshots.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts [hunk 1]
```diff
@@ -292,6 +287,19 @@ export const getStyles = memoizeFunction(
               display: 'inline-block',
               marginBottom: '8px'
             },
+            ':after': {
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: I'm good with these changes, the pseudo border is a pretty common approach for applying borders that don't affect the box model. I have two thoughts/concerns:

Can we use getFocusStyle here like on [line 125](https://github.com/OfficeDev/office-ui-fabric-react/blob/08490c6fd49de1ce064480049dbca3b5d5281e3c/packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts#L125)

And with this level of visual change, we need to over communicate for sure. Anyone that might be modifying the existing border styles (either for valid or invalid reasons) needs to know that these changes are coming.

**Comment by micahgodbolt** [comment]
Also, seems that screener is catching a bunch more changes that ben (or someone else) rejected. So seems to be a bit of cleanup there before we hit parity.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts (hunk 1 above)
Comment (betrue-final-final) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: Where would we communicate and how?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts [hunk 2]
```diff
@@ -292,6 +308,19 @@ export const getStyles = memoizeFunction(
...
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: Try commenting out this :after

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts [hunk 3]
```diff
@@ -24,10 +24,12 @@ const getDisabledStyles = memoizeFunction(
 
     return {
       backgroundColor: semanticColors.disabledBackground,
-      borderColor: semanticColors.disabledBackground,
       color: semanticColors.disabledText,
       cursor: 'default',
       selectors: {
+        ':after': {
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: Probably not needed now

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts [hunk 4]
```diff
@@ -167,9 +169,6 @@ export const getCaretDownButtonStyles = memoizeFunction(
         color: caret.buttonTextColor,
         fontSize: fonts.small.fontSize,
         position: 'absolute',
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: These lines from the original may need to come back since the main border isn't being changed now

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts [hunk 5]
```diff
@@ -276,22 +296,34 @@ export const getStyles = memoizeFunction(
...
-          borderColor: root.borderColor,
-          borderRadius: effects.roundedCorner2,
           cursor: 'text',
           display: 'block',
           height: ComboBoxHeight,
           overflow: 'hidden',
           whiteSpace: 'nowrap',
           textOverflow: 'ellipsis',
           boxSizing: 'border-box', // Border-box matches Dropdown and TextField
+          borderStyle: 'solid',
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: Need to bring back borderWidth too probably

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts [hunk 6]
```diff
@@ -303,40 +335,87 @@ export const getStyles = memoizeFunction(
       ],
 
       rootHovered: {
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.styles.ts]: Probably revert the changes to the rootHovered section (looks like those changes all pertain to the previous approach of using :after for all borders)

File: packages/date-time/src/components/DatePicker/DatePicker.styles.ts [hunk 7]
```diff
@@ -53,8 +53,7 @@ export const styles = (props: IDatePickerStyleProps): IDatePickerStyles => {
     callout: [classNames.callout],
     icon: [
       DatePickerIcon,
-      !label && [classNames.withoutLabel, { paddingTop: '7px' }],
-      label && [classNames.withLabel, { paddingBottom: '5px' }],
+      { paddingTop: '7px' },
```
[the remark this answers is not in the document]
Comment (betrue-final-final) [reply packages/date-time/src/components/DatePicker/DatePicker.styles.ts]: I'm not seeing withLabel or withoutLabel in types.ts for DatePicker in OUFR or date-time

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 8]
```diff
@@ -200,7 +200,10 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
             !disabled && rootHoverFocusActiveSelectorNeutralDarkMixin,
             {
               borderColor:
-                palette.themePrimary /* see https://github.com/OfficeDev/office-ui-fabric-react/pull/9182 for semantic color disc */
+                palette.themePrimary /* see https://github.com/OfficeDev/office-ui-fabric-react/pull/9182 for semantic color disc */,
```
[the remark this answers is not in the document]
Comment (betrue-final-final) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts]: I know, but it works!

File: packages/date-time/src/components/DatePicker/DatePicker.styles.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (ecraig12345) [reply packages/date-time/src/components/DatePicker/DatePicker.styles.ts]: Hmm it looks like they're from GlobalClassNames at the top of this file. So we should probably keep them for now, but never mind about the deprecation thing.

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts [hunk 9]
```diff
@@ -176,28 +176,52 @@ export const getStyles = memoizeFunction(
       labelDisabled: {},
       spinButtonWrapper: {
         display: 'flex',
+        position: 'relative',
         boxSizing: 'border-box',
         height: DEFAULT_HEIGHT,
         minWidth: DEFAULT_MIN_WIDTH,
-        border: `1px solid ${SpinButtonRootBorderColor}`,
-        borderRadius: effects.roundedCorner2
+        selectors: {
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: This section looks like it's left over from the previous approach

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts [hunk 10]
```diff
@@ -176,28 +176,52 @@ export const getStyles = memoizeFunction(
...
+          }
+        }
       },
       spinButtonWrapperTopBottom: {
         width: '100%'
       },
       spinButtonWrapperHovered: {
-        borderColor: SpinButtonRootBorderColorHovered,
         selectors: {
+          ':after': {
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: Same here

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts [hunk 11]
```diff
@@ -176,28 +176,52 @@ export const getStyles = memoizeFunction(
...
+            borderColor: SpinButtonRootBorderColorHovered
+          },
           [HighContrastSelector]: {
             borderColor: 'Highlight'
           }
         }
       },
       spinButtonWrapperFocused: {
-        borderColor: SpinButtonRootBorderColorFocused,
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: Probably bring this line back

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts [hunk 12]
```diff
@@ -176,28 +176,52 @@ export const getStyles = memoizeFunction(
...
+          },
+          ':after': {
+            pointerEvents: 'none',
+            content: "''",
+            position: 'absolute',
+            left: -1,
+            top: -1,
+            bottom: -1,
+            right: -1,
+            border: `2px solid ${SpinButtonRootBorderColorFocused}`,
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: Do you need a high contrast color for this too?

File: packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx [hunk 13]
```diff
@@ -332,8 +322,12 @@ export function getStyles(props: ITextFieldStyleProps): ITextFieldStyles {
       disabled && [
         {
           backgroundColor: semanticColors.disabledBackground,
-          borderColor: 'transparent',
-          color: semanticColors.disabledText
+          color: semanticColors.disabledText,
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.styles.tsx]: I think this section is left over from the previous approach

File: packages/date-time/src/components/DatePicker/DatePicker.styles.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (betrue-final-final) [reply packages/date-time/src/components/DatePicker/DatePicker.styles.ts]: Cool

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts (hunk 11 above)
Comment (betrue-final-final) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: This is in the :after

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts (hunk 12 above)
Comment (betrue-final-final) [reply packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts]: it's there.
